### DISCUSSION TITLE
Add remote random generated password to the GUI

### DISCUSSION THREAD

**Discussion by ghost** [agama-project/agama#1852] [Ideas]
Right now the only way to retreive the password for remote install is going into the tty, which isn't a newbie friendly approach imho. It wouls be nice to have the random password show up somewhere in the GUI installer, notifying the user how to remotely connect (agama.local) and what password to enter with. QR code would be cherry on the cake.

**Comment by jreidinger** [comment]
Well, this sadly a bit collide as when you need to use remote installation you do not have access to local GUI. What you usually have access to is either serial console or even not that. In such case you use PXE with password on kernel command line, so it is no longer random. Sadly I do not use case where you have local GUI and also remote one and use remote one.

**Reply by ghost** [reply]
Yeah it was just a wild idea, while I was playing around with my secondary PC, and I was logging into it from my remote main pc (like 10step away haha) and I was like, hmm, if I inserted my USB and booted from it, I wouldn't have to do all the kernel param or tty "hussle" if I'd just have the password show up on screen at step one, take a photo of it, bring myself over to my other cozy room and do the install remotely, now having the pw aswell. But just a wild idea yea

**Comment by dgdavid** [comment]
Thanks for the suggestion, @vajdao.

I can understand to some extend  the benefit you mentioned in terms of easing the remote access to the web client if the password were always random, specially while testing. However, as @jreidinger pointed out, it's difficult to think of a real use case that would justify such a feature, especially since the password can already be set using the `live_password` boot option when starting the ISO. For more details, see https://agama-project.github.io/docs/user/boot_options.

Moreover, the web UI is still just a client of Agama. As such, it should neither, have knowledge of the server password nor display it.

**Reply by ghost** [reply]
You're very welcome. It's okay, gotcha  :)
